Crash opening a non-existent Erlang file with ErlangLS

### Summary

Opening an Erlang file that does not exist crashes Helix. This appears to be caused by a bug in ErlangLS itself which causes the language server to crash because the file fails to open with `enoent` but Helix should probably attempt to handle language server shutdown gracefully.

As a workaround, one can `touch` the file before attempting to open it.

### Reproduction Steps

```
hx f.erl -vv
```

With `f.erl` not existing and `erlang_ls` installed.

<details><summary>Backtrace...</summary>

```
thread 'tokio-runtime-worker' panicked at 'pending_request with id not found!', helix-lsp/src/transport.rs:223:14
stack backtrace:
   0:     0x55badecb4194 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h606862f787600875
   1:     0x55bade5b9ebc - core::fmt::write::he803f0f418caf762
   2:     0x55badecac6ea - std::io::Write::write_fmt::h70bc45872f37e7bb
   3:     0x55badecb6f96 - std::panicking::default_hook::{{closure}}::hf51be35e2f510149
   4:     0x55badecb6cb2 - std::panicking::default_hook::h03ca0f22e1d2d25e
   5:     0x55badeb769bf - helix_term::application::Application::run::{{closure}}::{{closure}}::he4349962acdfbcae
   6:     0x55badecb77a8 - std::panicking::rust_panic_with_hook::h3b7380e99b825b63
   7:     0x55badecb7454 - std::panicking::begin_panic_handler::{{closure}}::h8e849d0710154ce0
   8:     0x55badecb46d4 - std::sys_common::backtrace::__rust_end_short_backtrace::hedcdaddbd4c46cc5
   9:     0x55badecb71bd - rust_begin_unwind
  10:     0x55bade542413 - core::panicking::panic_fmt::he1bbc7336d49a357
  11:     0x55bade5b8581 - core::panicking::panic_display::hb184394d54dc8360
  12:     0x55bade5b852b - core::panicking::panic_str::ha42887cd972750c9
  13:     0x55bade542286 - core::option::expect_failed::h9159c9d16ed9fb7c
  14:     0x55bade7c7bff - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h14bb53a343d9aa39
  15:     0x55bade7c5493 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h0c16f179a49dfae4
  16:     0x55bade7b0b1b - std::panic::catch_unwind::h897cdf4ea94d5591
  17:     0x55bade7ef383 - tokio::runtime::task::harness::poll_future::h06af2fb768c08003
  18:     0x55bade7eb6e4 - tokio::runtime::task::harness::Harness<T,S>::poll::h605a328364e8efe4
  19:     0x55badecfbd30 - std::thread::local::LocalKey<T>::with::hfd1572d00b8740ce
  20:     0x55badecf3453 - tokio::runtime::thread_pool::worker::Context::run_task::h494e0bbdf818ef37
  21:     0x55badecf2c0f - tokio::runtime::thread_pool::worker::Context::run::h55223f5e876e0d9f
  22:     0x55badecdc63b - tokio::macros::scoped_tls::ScopedKey<T>::set::h29bdcf4fdeabe957
  23:     0x55badecf209c - tokio::runtime::thread_pool::worker::run::h360ffc7394766e08
  24:     0x55badecd5b71 - <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll::h277715bb664d31a3
  25:     0x55badecd7451 - tokio::runtime::task::harness::Harness<T,S>::poll::h18a98004e12cec29
  26:     0x55badece0775 - tokio::runtime::blocking::pool::Inner::run::hdaa78bfe5b72badb
  27:     0x55badecde69c - std::sys_common::backtrace::__rust_begin_short_backtrace::hc55fdb9d74d65b0e
  28:     0x55badece7a53 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h1be911bb0b429398
  29:     0x55badecbdd95 - std::sys::unix::thread::Thread::new::thread_start::ha07928d93d5a5ec9
  30:     0x7f12a2c82eb2 - start_thread
  31:     0x7f12a2d0531c - clone3
  32:                0x0 - <unknown>
```

</details>

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-05-14T17:47:13.842 helix_loader [DEBUG] Located configuration folders: []
2022-05-14T17:47:13.964 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":false},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"publishDiagnostics":{},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":false}},"window":{"workDoneProgress":true},"workspace":{"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"workspaceFolders":true}},"processId":31320,"rootUri":null,"workspaceFolders":[]},"id":0}
2022-05-14T17:47:14.300 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.288948-05:00] [info] Started erlang_ls server [erlang_ls:main/1 L29] <0.9.0>\n","type":4}}
2022-05-14T17:47:14.300 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.288948-05:00] [info] Started erlang_ls server [erlang_ls:main/1 L29] <0.9.0>\n" }
2022-05-14T17:47:14.420 helix_lsp::transport [INFO] <- {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"callHierarchyProvider":true,"codeActionProvider":true,"codeLensProvider":{"resolveProvider":false},"completionProvider":{"resolveProvider":true,"triggerCharacters":[":","#","?",".","-","\""]},"definitionProvider":true,"documentFormattingProvider":true,"documentHighlightProvider":true,"documentRangeFormattingProvider":false,"documentSymbolProvider":true,"executeCommandProvider":{"commands":["31329:server-info","31329:ct-run-test","31329:show-behaviour-usages","31329:suggest-spec","31329:function-references"]},"foldingRangeProvider":true,"hoverProvider":true,"implementationProvider":true,"referencesProvider":true,"renameProvider":true,"textDocumentSync":{"change":2,"openClose":true,"save":{"includeText":false}},"workspaceSymbolProvider":true},"serverInfo":{"name":"Erlang LS","version":"0.0.0"}}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"capabilities":{"callHierarchyProvider":true,"codeActionProvider":true,"codeLensProvider":{"resolveProvider":false},"completionProvider":{"resolveProvider":true,"triggerCharacters":[":","#","?",".","-","\""]},"definitionProvider":true,"documentFormattingProvider":true,"documentHighlightProvider":true,"documentRangeFormattingProvider":false,"documentSymbolProvider":true,"executeCommandProvider":{"commands":["31329:server-info","31329:ct-run-test","31329:show-behaviour-usages","31329:suggest-spec","31329:function-references"]},"foldingRangeProvider":true,"hoverProvider":true,"implementationProvider":true,"referencesProvider":true,"renameProvider":true,"textDocumentSync":{"change":2,"openClose":true,"save":{"includeText":false}},"workspaceSymbolProvider":true},"serverInfo":{"name":"Erlang LS","version":"0.0.0"}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.310922-05:00] [info] Reading config file. path=\"/home/michael/.config/erlang_ls/erlang_ls.config\" [els_config:consult_config/1 L265] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.318716-05:00] [info] Reading config file. path=\"/home/michael/erlang_ls.config\" [els_config:consult_config/1 L265] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.321731-05:00] [info] Reading config file. path=\"/home/michael/erlang_ls.yaml\" [els_config:consult_config/1 L265] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.310922-05:00] [info] Reading config file. path=\"/home/michael/.config/erlang_ls/erlang_ls.config\" [els_config:consult_config/1 L265] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.321828-05:00] [info] No config file found. [els_config:consult_config/1 L256] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/showMessage","params":{"message":"The current project is missing an erlang_ls.config file. Need help configuring Erlang LS for your project? Visit: https://erlang-ls.github.io/configuration/","type":2}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.321896-05:00] [info] OTP Path: \"/nix/store/l9s56l837qpvnpqsf5hnkmi0gba7abh5-erlang-24.1.6/lib/erlang\" [els_config:do_initialize/4 L110] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.351048-05:00] [info] Excluded OTP Applications: [\"megaco\",\"diameter\",\"snmp\",\"wx\"] [els_config:do_initialize/4 L125] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"[2022-05-14T17:47:14.351161-05:00] [info] Config=#{\"apps_dirs\" => [\"_build/default/lib/*\"],\"include_dirs\" => [\"_build/default/lib/*/include\",\"include\"]} [els_config:do_initialize/4 L154] <0.150.0>\n","type":4}}
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.318716-05:00] [info] Reading config file. path=\"/home/michael/erlang_ls.config\" [els_config:consult_config/1 L265] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.321731-05:00] [info] Reading config file. path=\"/home/michael/erlang_ls.yaml\" [els_config:consult_config/1 L265] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.321828-05:00] [info] No config file found. [els_config:consult_config/1 L256] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_term::application [WARN] unhandled window/showMessage: ShowMessageParams { typ: Warning, message: "The current project is missing an erlang_ls.config file. Need help configuring Erlang LS for your project? Visit: https://erlang-ls.github.io/configuration/" }
2022-05-14T17:47:14.421 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"erlang","text":"\n","uri":"file:///home/michael/f.erl","version":0}}}
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.321896-05:00] [info] OTP Path: \"/nix/store/l9s56l837qpvnpqsf5hnkmi0gba7abh5-erlang-24.1.6/lib/erlang\" [els_config:do_initialize/4 L110] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.351048-05:00] [info] Excluded OTP Applications: [\"megaco\",\"diameter\",\"snmp\",\"wx\"] [els_config:do_initialize/4 L125] <0.150.0>\n" }
2022-05-14T17:47:14.421 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Log, message: "[2022-05-14T17:47:14.351161-05:00] [info] Config=#{\"apps_dirs\" => [\"_build/default/lib/*\"],\"include_dirs\" => [\"_build/default/lib/*/include\",\"include\"]} [els_config:do_initialize/4 L154] <0.150.0>\n" }
2022-05-14T17:47:14.477 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/showMessage","params":{"message":"Erlang LS (in michael), version: 0.0.0, OTP version: 24","type":3}}
2022-05-14T17:47:14.477 helix_term::application [WARN] unhandled window/showMessage: ShowMessageParams { typ: Info, message: "Erlang LS (in michael), version: 0.0.0, OTP version: 24" }
2022-05-14T17:47:14.485 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.484015-05:00] [error] Generic server els_provider terminating. Reason: {{badmatch,{error,enoent}},[{els_indexing,shallow_index,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_indexing.erl\"},{line,145}]},{els_utils,lookup_document,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_utils.erl\"},{line,158}]},{els_text_synchronization,did_open,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization.erl\"},{line,52}]},{els_text_synchronization_provider,handle_request,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization_provider.erl\"},{line,25}]},{els_provider,handle_call,3,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_provider.erl\"},{line,107}]},{gen_server,try_handle_call,4,[{file,\"gen_server.erl\"},{line,721}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}. Last message: {handle_request,els_text_synchronization_provider,{did_open,#{<<\"textDocument\">> => #{<<\"languageId\">> => <<\"erlang\">>,<<\"text\">> => <<\"\\n\">>,<<\"uri\">> => <<\"file:///home/michael/f.erl\">>,<<\"version\">> => 0}}}}. State: #{in_progress => [],in_progress_diagnostics => []}. Client els_server stacktrace: [{gen,do_call,4,[{file,\"gen.erl\"},{line,214}]},{gen_server,call,3,[{file,\"gen_server.erl\"},{line,243}]},{els_methods,textdocument_didopen,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_methods.erl\"},{line,191}]},{els_methods,dispatch,4,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_methods.erl\"},{line,74}]}]. [gen_server:error_info/7 L949] <0.150.0>\n"
2022-05-14T17:47:14.485 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.484723-05:00] [error] crasher: initial call: els_provider:init/1, pid: <0.150.0>, registered_name: els_provider, error: {{badmatch,{error,enoent}},[{els_indexing,shallow_index,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_indexing.erl\"},{line,145}]},{els_utils,lookup_document,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_utils.erl\"},{line,158}]},{els_text_synchronization,did_open,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization.erl\"},{line,52}]},{els_text_synchronization_provider,handle_request,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization_provider.erl\"},{line,25}]},{els_provider,handle_call,3,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_provider.erl\"},{line,107}]},{gen_server,try_handle_call,4,[{file,\"gen_server.erl\"},{line,721}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}, ancestors: [els_sup,<0.136.0>], message_queue_len: 0, messages: [], links: [<0.137.0>], dictionary: [], trap_exit: true, status: running, heap_size: 28690, stack_size: 29, reductions: 3471764; neighbours: [proc_lib:crash_report/4 L525] <0.150.0>\n"
2022-05-14T17:47:14.486 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.485391-05:00] [error] Supervisor: {local,els_sup}. Context: child_terminated. Reason: {{badmatch,{error,enoent}},[{els_indexing,shallow_index,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_indexing.erl\"},{line,145}]},{els_utils,lookup_document,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_utils.erl\"},{line,158}]},{els_text_synchronization,did_open,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization.erl\"},{line,52}]},{els_text_synchronization_provider,handle_request,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization_provider.erl\"},{line,25}]},{els_provider,handle_call,3,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_provider.erl\"},{line,107}]},{gen_server,try_handle_call,4,[{file,\"gen_server.erl\"},{line,721}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}. Offender: id=els_provider,pid=<0.150.0>. [supervisor:do_restart/3 L736] <0.137.0>\n"
2022-05-14T17:47:14.486 helix_lsp::transport [INFO] <- {"error":{"code":-32001,"message":"Unexpected error while textDocument/didOpen"},"id":null,"jsonrpc":"2.0"}
2022-05-14T17:47:14.486 helix_lsp::transport [ERROR] <- Server error: Unexpected error while textDocument/didOpen
2022-05-14T17:47:14.488 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.485374-05:00] [error] Unexpected error [type=exit] [error={{{badmatch,{error,enoent}},[{els_indexing,shallow_index,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_indexing.erl\"},{line,145}]},{els_utils,lookup_document,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_utils.erl\"},{line,158}]},{els_text_synchronization,did_open,1,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization.erl\"},{line,52}]},{els_text_synchronization_provider,handle_request,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_text_synchronization_provider.erl\"},{line,25}]},{els_provider,handle_call,3,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_core/src/els_provider.erl\"},{line,107}]},{gen_server,try_handle_call,4,[{file,\"gen_server.erl\"},{line,721}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]},{gen_server,call,[els_provider,{handle_request,els_text_synchronization_provider,{did_open,#{<<\"textDocument\">> => #{<<\"languageId\">> => <<\"erlang\">>,<<\"text\">> => <<\"\\n\">>,<<\"uri\">> => <<\"file:///home/michael/f.erl\">>,<<\"version\">> => 0}}}},infinity]}}] [stack=[{gen_server,call,3,[{file,\"gen_server.erl\"},{line,247}]},{els_methods,textdocument_didopen,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_methods.erl\"},{line,191}]},{els_methods,dispatch,4,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_methods.erl\"},{line,74}]},{els_server,handle_request,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_server.erl\"},{line,162}]},{lists,foldl,3,[{file,\"lists.erl\"},{line,1267}]},{els_server,handle_cast,2,[{file,\"/build/f6zcfzjgac200djpxhsg5cs77gbz9sml-source/apps/els_lsp/src/els_server.erl\"},{line,119}]},{gen_server,try_dispatch,4,[{file,\"gen_server.erl\"},{line,695}]},{gen_server,handle_msg,6,[{file,\"gen_server.erl\"},{line,771}]}]] [els_methods:dispatch/4 L79] <0.148.0>\n"
2022-05-14T17:47:14.526 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.525676-05:00] [error] Supervisor: {<0.63.0>,user_sup}. Context: child_terminated. Reason: epipe. Offender: pid=<0.64.0>,mod=user_sup. [supervisor_bridge:report_error/3 L149] <0.63.0>\n"
2022-05-14T17:47:14.526 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.525855-05:00] [error] Generic server <0.63.0> terminating. Reason: epipe. Last message: {'EXIT',<0.64.0>,epipe}. State: {state,user_sup,undefined,<0.64.0>,{<0.63.0>,user_sup}}. [gen_server:error_info/7 L949] <0.63.0>\n"
2022-05-14T17:47:14.526 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.526058-05:00] [error] crasher: initial call: supervisor_bridge:user_sup/1, pid: <0.63.0>, registered_name: [], exit: {epipe,[{gen_server,handle_common_reply,8,[{file,\"gen_server.erl\"},{line,811}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]}, ancestors: [kernel_sup,<0.47.0>], message_queue_len: 0, messages: [], links: [<0.49.0>], dictionary: [], trap_exit: true, status: running, heap_size: 4185, stack_size: 29, reductions: 36097; neighbours: [proc_lib:crash_report/4 L525] <0.63.0>\n"
2022-05-14T17:47:14.526 helix_lsp::transport [ERROR] err <- "[2022-05-14T17:47:14.526434-05:00] [error] Supervisor: {local,kernel_sup}. Context: child_terminated. Reason: epipe. Offender: id=user,pid=<0.63.0>. [supervisor:do_restart/3 L751] <0.49.0>\n"
2022-05-14T17:47:17.479 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":0,"line":1},"start":{"character":0,"line":0}},"text":"\n"}],"textDocument":{"uri":"file:///home/michael/f.erl","version":1}}}
2022-05-14T17:47:19.105 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"shutdown","params":null,"id":1}
2022-05-14T17:47:19.607 helix_term::application [ERROR] Timed out waiting for language servers to shutdown
```

</details>


### Platform

Linux (NixOS 21.11)

### Terminal Emulator

kitty 0.23.1

### Helix Version

22.03-209-g823eaad1

--- a/helix-lsp/src/transport.rs
+++ b/helix-lsp/src/transport.rs
@@ -215,20 +215,21 @@ impl Transport {
             }
         };
 
-        let tx = self
-            .pending_requests
-            .lock()
-            .await
-            .remove(&id)
-            .expect("pending_request with id not found!");
-
-        match tx.send(result).await {
-            Ok(_) => (),
-            Err(_) => error!(
-                "Tried sending response into a closed channel (id={:?}), original request likely timed out",
-                id
-            ),
-        };
+        if let Some(tx) = self.pending_requests.lock().await.remove(&id) {
+            match tx.send(result).await {
+                Ok(_) => (),
+                Err(_) => error!(
+                    "Tried sending response into a closed channel (id={:?}), original request likely timed out",
+                    id
+                ),
+            };
+        } else {
+            log::error!(
+                "Discarding Language Server response without a request (id={:?}) {:?}",
+                id,
+                result
+            );
+        }
 
         Ok(())
     }

